src/Bicep.Core/TypeSystem/Providers/ThirdParty/ThirdPartyResourceTypeProvider.cs

```diff
@@ -0,0 +1,120 @@
+                // TODO: remove 'dependsOn' from the type library
+                properties = properties.SetItem(LanguageConstants.ResourceDependsOnPropertyName, new TypeProperty(LanguageConstants.ResourceDependsOnPropertyName, LanguageConstants.ResourceOrResourceCollectionRefArray, TypePropertyFlags.WriteOnly | TypePropertyFlags.DisallowAny));
+            }
+
+            return new ObjectType(
+                objectType.Name,
+                objectType.ValidationFlags,
+                isExistingResource ? ConvertToReadOnly(properties.Values) : properties.Values,
+                objectType.AdditionalPropertiesType,
+                isExistingResource ? ConvertToReadOnly(objectType.AdditionalPropertiesFlags) : objectType.AdditionalPropertiesFlags,
+                functions: null);
+        }
+
+        private static IEnumerable<TypeProperty> ConvertToReadOnly(IEnumerable<TypeProperty> properties)
+        {
+            foreach (var property in properties)
+            {
+                yield return new TypeProperty(property.Name, property.TypeReference, ConvertToReadOnly(property.Flags), property.Description);
+            }
+        }
+
+        private static TypePropertyFlags ConvertToReadOnly(TypePropertyFlags typePropertyFlags)
+            => (typePropertyFlags | TypePropertyFlags.ReadOnly) & ~TypePropertyFlags.Required;
+
+        public ResourceType? TryGetDefinedType(NamespaceType declaringNamespace, ResourceTypeReference typeReference, ResourceTypeGenerationFlags flags)
+        {
+            if (!HasDefinedType(typeReference))
+            {
+                return null;
+            }
+
+            // It's important to cache this result because generating the resource type is an expensive operation
+            var resourceType = definedTypeCache.GetOrAdd(flags, typeReference, () =>
+            {
+                var resourceType = resourceTypeLoader.LoadType(typeReference);
+
+                return SetBicepResourceProperties(resourceType, flags);
+            });
+
+            return new(
+                declaringNamespace,
+                resourceType.TypeReference,
+                resourceType.ValidParentScopes,
+                resourceType.ReadOnlyScopes,
+                resourceType.Flags,
+                resourceType.Body,
+                ImmutableHashSet<string>.Empty);
+        }
+
+        //TODO - We're not ready to add fallback types, as this requires a change to the types.json package first
+        public ResourceType? TryGenerateFallbackType(NamespaceType declaringNamespace, ResourceTypeReference typeReference, ResourceTypeGenerationFlags flags)
+        {
+            return null;
+        }
+
+        public bool HasDefinedType(ResourceTypeReference typeReference)
+            => availableResourceTypes.Contains(typeReference);
+
+        public IEnumerable<ResourceTypeReference> GetAvailableTypes()
+            => availableResourceTypes;
+
+        public string Version { get; } = "1.0.0";
```

Comment:
The version should be coming from the registry path, not hard-coded to `1.0.0`

Reply:
Ah yes, forget to change